fuel-merkle/src/sparse/proof.rs

```diff
@@ -58,9 +58,15 @@ pub struct InclusionProof {
 impl InclusionProof {
     pub fn verify(&self, root: &Bytes32, key: &MerkleTreeKey, value: &[u8]) -> bool {
         let Self { proof_set } = self;
+
+        if proof_set.len() > u32::MAX as usize {
```

Comment:
Good catch! We can actually make this 
```
if proof_set.len() > 256 {
```
since a valid proof set cannot have more than log(n) side nodes.